File: server/app/models/company_location.rb
class CompanyLocation < ActiveRecord::Base
	acts_as_mappable

	has_many :jobs
	belongs_to :company

	accepts_nested_attributes_for :jobs

	validates_presence_of :title, :email, :city, :lat, :lng

	after_save :create_slug

  def to_param
    [id, title.parameterize].join("-")
  end

	def create_slug
		self.update_column(:slug, self.to_param)
	end
end

File: client/app/models/company-location.js
import DS from 'ember-data';

export default DS.Model.extend({
	title: DS.attr('string'),
	email: DS.attr('string'),
  city: DS.attr('string'),
  county: DS.attr('string'),
  country: DS.attr('string'),
  lat: DS.attr(''),
  lng: DS.attr(''),
  company: DS.belongsTo('company'),
  jobs: DS.hasMany('job', { async: true })
});
